.github/workflows/build.yml

```diff
@@ -0,0 +1,95 @@
+name: Build and Deploy
+
+on:
+  push:
+
+permissions:
+  contents: write
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, ubuntu-24.04-arm, windows-latest, macos-13, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with: { python-version: "3.12" }
+
+      - name: Build wheels with cibuildwheel
+        uses: pypa/cibuildwheel@v3.1.4
+        with: { output-dir: wheelhouse }
+        env:
+          CIBW_SKIP: "*-win32"
+          CIBW_BEFORE_BUILD: "python -m pip install --upgrade pip setuptools wheel"
+          CIBW_TEST_REQUIRES: "pytest"
+          CIBW_TEST_COMMAND: "pytest -v {project}/tests"
+
+      - name: Upload wheel artifacts
+        if: ${{ github.ref_type == 'tag' && startsWith(github.ref_name, 'v') }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: cibw-wheels-${{ matrix.os }}-${{ strategy.job-index }}
+          path: wheelhouse/*.whl
+
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with: { python-version: "3.12" }
+      - name: Build sdist
+        run: |
+          python -m pip install --upgrade pip build
+          python -m build --sdist
+
+      # Only upload on tag pushes
+      - name: Upload sdist
+        if: ${{ github.ref_type == 'tag' && startsWith(github.ref_name, 'v') }}
```

Comment:
I think here as well, we should always upload the artifact for the distribution.